core/mutator/entry/entry.go

```diff
@@ -16,12 +16,43 @@
 package entry
 
 import (
+	"fmt"
+
 	"github.com/golang/glog"
 	"github.com/golang/protobuf/proto"
 
 	pb "github.com/google/keytransparency/core/api/v1/keytransparency_go_proto"
+	tpb "github.com/google/trillian"
 )
 
+func FromMapLeaf(leaf *tpb.MapLeaf) (*pb.EntryUpdate, error) {
```

Comment:
[nit, not blocking] some of the naming here still feels a little uncomfortable. It feels a little strange to have a method to turn a map leaf into an "update". An Update feels like something that would wrap a logical Value type until it is applied to the map. Once it is applied to the Map I would expect to be talking about Leaves/Nodes and the Value type.

Reply:
The types are admittedly non-ideal. Trying to work toward something better. 

## Current State
`LogItem` is currently an `EntryUpdate`. 
`LogItems` get mapped to `IndexedUpdates` - Perhaps there's a better name / type for this. 
`IndexedUpdates` get grouped by index 
`EntryUpdate` + `[]EntryUpdate` gets combined `EntryUpdate`.
The reducer adds the index back, to get `IndexedUpdate` 
In the final step `IndexedUpdate` gets converted to `MapLeaf`.

## Future State?
`LogItem` is an `EntryUpdate`.  (Also an IndexedValue?) 
`LogItems` get mapped to `???` / `IndexedValue`
`IndexedValue`s get grouped by index 
`Value` + `[]Value` gets combined to a new `Value`.
*Question: should we be reducing `[]Value` or `[]IndexedValue` ?*
The reducer adds the index back, to get `IndexedValue` 
In the final step `IndexedValue` gets converted to `MapLeaf`.

Reply:
I'm happy reducing an IndexedValue. I can imagine some situations where the key might be relevant.

Your proposed changes sound much clearer to me. Thanks for the clarification :-)

Reply:
Moved to IndexedValue